### PR TITLE
[Split_1]template change for split

### PR DESCRIPTION
**Pull request by vicancy**
This PR contains template change when pages are splitted into overload level. For overload pages, `type` equals to `OverloadGroup`. Note that default template is also modified to be compatible with overload pages, with the following consideration
a. The added logic does not influence current logic, and it makes sense for `MREF.common` to be compatible with multiple scenarios.
b. Template code inside plugin can be really hard to maintain if it provides another copy of `ManagedReference.common.js`.

### CODE REVIEW

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/resources/partials/class.tmpl.partial [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  <tbody>
+  {{/children.0}}
+    {{#children}}
+    <tr>
+      <td id="{{id}}" data-uid="{{uid}}">
+        <xref uid="{{uid}}" altProperty="fullName" displayProperty="name"/>
+      </td>
+      <td class="markdown level1 summary">{{{summary}}}</td>
+    </tr>
+    {{/children}}
```
Comment (superyyrrzz) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/resources/partials/class.tmpl.partial]: >     [](start = 17, length = 4)

remove

File: src/docfx.website.themes/default/partials/title.tmpl.partial [hunk 2]
```diff
@@ -16,4 +16,25 @@ Enum {{name.0.value}}
...
+Method {{name.0.value}}
+{{/inMethod}}
+{{#inEvent}}
+Event {{name.0.value}}
+{{/inEvent}}
+{{#inOperator}}
+Operator {{name.0.value}}
+{{/inOperator}}
+{{#inEii}}
+Eii {{name.0.value}}
```
Comment (superyyrrzz) [inline src/docfx.website.themes/default/partials/title.tmpl.partial]: >Eii [](start = 0, length = 3)

to full name

File: src/docfx.website.themes/common/ManagedReference.common.js [hunk 3]
```diff
@@ -43,6 +43,19 @@ exports.transform = function (model)  {
...
+        model[classItems[model.memberType.toLowerCase()].typePropertyName] = true;
+        if (model.children) groupChildren(model, classItems);
+        break;
+      case 'field':
+        model.isNonOverload = true;
+        model.inField = true;
+        break;
+      case 'event':
+        model.isNonOverload = true;
+        model.inEvent = true;
```
Comment (superyyrrzz) [inline src/docfx.website.themes/common/ManagedReference.common.js]: move to mref's postTransform?

File: src/docfx.website.themes/default/ManagedReference.html.primary.tmpl [hunk 4]
```diff
@@ -48,6 +48,12 @@
             {{#isEnum}}
               {{>partials/enum}}
             {{/isEnum}}
+            {{#isOverload}}
+              {{>partials/overload}}
+            {{/isOverload}}
+            {{#isNonOverload}}
+              {{>partials/nonOverload}}
+            {{/isNonOverload}}
```
Comment (superyyrrzz) [inline src/docfx.website.themes/default/ManagedReference.html.primary.tmpl]: isExtension?

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/resources/toc.html.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                grouped[type].push(element);
+            } else {
+                items.push(element);
+            }
+        }, this);
+        for (var key in grouped) {
+            if (grouped.hasOwnProperty(key)) {
+                items.push({
+                    name: model.__global[groupNames[key].key] || groupNames[key].name,
+                    items: grouped[key]
```
Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/resources/toc.html.js]: >items [](start = 20, length = 5)

minor: should it name with "item"?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
>   if (category === 'class') {

update with classCategory?

---
Refers to: src/docfx.website.themes/common/ManagedReference.common.js:175 in eec1e73. [](commit_id = eec1e73dec55a8cdded3978d67953981d368d577, deletion_comment = False)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**